### DISCUSSION TITLE
What can I do with a node id returned from getNodeId?

### DISCUSSION THREAD

**Discussion by JordanSalinas** [mobxjs/mobx-state-tree#1876] [Q&A]
What's the purpose of getNodeId? Currently I'm using it in React as a key when iterating lists and returning components. What I'd like to do is use it to modify data in the store. i.e. removeNode(nodeId). Is there a way to use it in this way? Honestly even if I could just do getNodeById(nodeId) that'd be useful as well because then I could just use the built in methods like destroy(node). I suppose I could just implement this myself, but I think having a getNodeById method would be a lot easier and cleaner.

**Comment by EmilTholin** [comment]
Hi @JordanSalinas!

Do you think you could include a code example of where you are experiencing this? Maybe you could approach the problem from another angle and use e.g. a [`reference`](https://mobx-state-tree.js.org/concepts/references).

**Reply by JordanSalinas** [reply]
I've considered references, however the problem stems from newly created content. I don't have an identifier until the data is persisted to the backend. I'm working with a tree of data, but for simplicity imagine I can create a list of objects and none of these objects have id's because they haven't yet been persisted. Not only do references not work here, but when trying to modify the array it's difficult to find the item in the list because again, no identifier. 

Obviously I could create a temporary identifier myself, but this adds complexity where I think using a combination of getNodeId(target) and then getNodeById(nodeId) would help and be a lot cleaner.

I suppose that's all more of a feature request, but it made me wonder what else I could actually use getNodeId for.
